app/models/agents/post_agent.rb

```diff
@@ -124,6 +137,33 @@ def check
 
     private
 
+    def normalize_response_headers(headers)
+      case interpolated['event_headers_style']
+      when nil, '', 'capitalized'
+        normalize = ->name {
+          name.gsub(/(?:\A|(?<=-))([[:alpha:]])|([[:alpha:]]+)/) {
+            $1 ? $1.upcase : $2.downcase
+          }
+        }
+      when 'downcased'
+        normalize = :downcase.to_proc
+      when 'snakecased', nil
+        normalize = ->name { name.tr('A-Z-', 'a-z_') }
+      when 'raw'
+        if Object.method_defined?(:itself)  # Ruby >= 2.2
+          normalize = :itself.to_proc
+        else
+          normalize = ->name { name }
```

Comment:
Yes, I was just so much looking forward to using the new features that will become the standard when Rails 5 (that requires Ruby >=2.2) launches.  Will change it before merging this PR.